app/front/scripts/services/upload-file.js

```diff
@@ -35,7 +35,7 @@ angular.module('Application')
               status.sampleSize += 1;
             }
 
-            if (!status.errors) {
+            if (status.report.valid) {
```

Comment:
From what I understood, this will fail if there was an error getting the report, as `status.report` will be `undefined`, and `status.report.valid` will raise an exception. We need to check for `status.state === 'completed'`, like you do in the templates. Better yet, we can simply test for `status.report` being defined, this way the dependency between the codebases are simply on `status.report`, and not `status.state`.